### PR TITLE
New species lozenge

### PR DESCRIPTION
**Pull request by jyothishnt**
## Type

- [ ] Bug fix
- [ ] New feature
- [x] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-752

## Description
Creating a new selected species component which will eventually replace the current one
Also added a storybook entry

## Views affected
Nothing atm as we are not using it anywhere. But will affect all the apps once we start using this component.
We intend to test it on the new species homepage app

- [ ] I have checked any requirements for Google Analytics
- [x] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

## Possible complications
Nothing I am aware.

### CODE REVIEW

File: src/ensembl/src/shared/components/new-selected-species/selected-species-common.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
+@import 'src/styles/common';
+
+$selectedSpeciesBorderRadius: 20px;
+$selectedSpeciesPadding: 6px 20px;
+$deleteButtonPadding: 8px 20px 8px 28px;
+
+@mixin selected-species-common {
```
Comment (azangru) [inline src/ensembl/src/shared/components/new-selected-species/selected-species-common.scss:7]: this mixin (and the file itself) probably don't make sense on their own anymore; previously, there were simple selected species and focusable selected species, both of which were using styles from this common file; but now there's only one component; so these styles can move over to `SelectedSpecies.scss`

File: src/ensembl/src/shared/components/new-selected-species/SelectedSpecies.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    props.onMouseEnter && props.onMouseEnter();
+  };
+
+  const handleMouseLeave = () => {
+    props.onMouseLeave && props.onMouseLeave();
+  };
+
+  const handleClick = () => {
+    if (!props.isActive) {
+      props.onClick(props.species.genome_id);
```
Comment (azangru) [inline src/ensembl/src/shared/components/new-selected-species/SelectedSpecies.tsx:63]: After Andrea's updated design, I don't know if it is still correct. Andrea is not sure either. Much indecision here...

File: src/ensembl/src/shared/components/new-selected-species/SelectedSpecies.test.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    it('is inactive and disabled', () => {
+      propsObj.species.isEnabled = false;
+      propsObj.isActive = false;
+      const wrapper = renderSelectedSpecies(propsObj);
+      expect(wrapper.children('div').hasClass('notInUseInactive')).toEqual(
+        true
+      );
+    });
+  });
+});
```
Comment (azangru) [inline src/ensembl/src/shared/components/new-selected-species/SelectedSpecies.test.tsx:73]: Whatever we decide about the onClick behaviour, we had better include it in the tests.

File: src/ensembl/stories/shared-components/selected-species/SelectedSpecies.stories.scss [hunk 4]
```diff
@@ -0,0 +1,5 @@
+@import 'src/styles/common';
+
+.wrapper {
+  padding: 40px;
+}
```
Comment (azangru) [inline src/ensembl/stories/shared-components/selected-species/SelectedSpecies.stories.scss:5]: new line, please

File: src/ensembl/src/shared/components/new-selected-species/selected-species-common.scss (hunk 1 above)
Comment (jyothishnt) [reply src/ensembl/src/shared/components/new-selected-species/selected-species-common.scss:7]: We need to create a ticket to create new mixin that can be used for round buttons (as decided in the code review)

File: src/ensembl/src/shared/components/new-selected-species/SelectedSpecies.tsx (hunk 2 above)
Comment (jyothishnt) [reply src/ensembl/src/shared/components/new-selected-species/SelectedSpecies.tsx:63]: Didnt we decide there is no change in this functionality?

File: src/ensembl/src/shared/components/new-selected-species/selected-species-common.scss (hunk 1 above)
Comment (azangru) [reply src/ensembl/src/shared/components/new-selected-species/selected-species-common.scss:7]: https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-774

No need to create a new one 😄

File: src/ensembl/src/shared/components/new-selected-species/SelectedSpecies.tsx (hunk 2 above)
Comment (azangru) [reply src/ensembl/src/shared/components/new-selected-species/SelectedSpecies.tsx:63]: No change. Should probably demonstrate it in a test.

File: src/ensembl/src/shared/components/new-selected-species/SelectedSpecies.test.tsx (hunk 3 above)
Comment (jyothishnt) [reply src/ensembl/src/shared/components/new-selected-species/SelectedSpecies.test.tsx:95]: this is done.

File: src/ensembl/stories/shared-components/selected-species/SelectedSpecies.stories.scss (hunk 4 above)
Comment (jyothishnt) [reply src/ensembl/stories/shared-components/selected-species/SelectedSpecies.stories.scss]: done

**Comment by azangru** [comment]
Pushed commit https://github.com/Ensembl/ensembl-client/pull/350/commits/d4f5fb47150b5e8f909225a56933658dbf367ad4 to the PR. Most importantly, it's about not mutating the globally defined props object between tests. I've also updated some names.

**Review by azangru — APPROVED** [review]

**Pull request merged**